### DISCUSSION TITLE
"The trait bound by derive is not satisfied" in IDE, but still can be compiled.

### DISCUSSION THREAD

**Discussion by Parallel-Paradox** [bevyengine/bevy#5755] [Q&A]
![image](https://user-images.githubusercontent.com/81791934/185778683-de6edd62-217b-46e9-bf90-67b09ec91a54.png)

And the warning is:

![image](https://user-images.githubusercontent.com/81791934/185778700-5546deea-16d4-4c8d-8651-8c678c8995e8.png)

Does this issue come from rust plugin of CLion?

**Comment by afonsolage** [comment]
![image](https://user-images.githubusercontent.com/1176452/185778982-eb48e657-4c52-400a-b0da-952b486eb73f.png)

Tested on Bevy `0.8.0` and it seems to work fine. 

![image](https://user-images.githubusercontent.com/1176452/185779027-7c2c6d12-2534-49c9-b167-bc3793063053.png)

If you just do a `cargo check`, it doesn't shows those errors? If `cargo check` doesn't report anything, so it must be something with your IDE setup.

**Comment by Parallel-Paradox** [comment]
Yes, it actually works well in vscode, but not in clion with rust plugin I guess. The derive feature may not be supported correctly.
Looks like rust_analysis is still the best support.

**Reply by MrGVSV** [reply]
Do you have the experimental features on in CLion? It _might_ be related to that.

**Reply by Parallel-Paradox** [reply]
> Do you have the experimental features on in CLion? It _might_ be related to that.

I tried to trun on/off all the experimental features, nothing changed.

**Comment by Parallel-Paradox** [comment]
Post an [issue](https://github.com/intellij-rust/intellij-rust/issues/9226#issue-1346523801) in rust plugin.

**Comment by Parallel-Paradox** [comment]
Yes, it actually works well in vscode, but not in clion with rust plugin I guess. The derive feature may not be supported correctly.

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Afonso Lage ***@***.***>
发送时间: Sunday, August 21, 2022 2:40:56 PM
收件人: bevyengine/bevy ***@***.***>
抄送: Parallel-Paradox ***@***.***>; Author ***@***.***>
主题: Re: [bevyengine/bevy] "The trait bound by derive is not satisfied" in IDE, but still can be compiled. (Discussion #5755)


[image]<https://user-images.githubusercontent.com/1176452/185778982-eb48e657-4c52-400a-b0da-952b486eb73f.png>

Tested on Bevy 0.8.0 and it seems to work fine.

If you just do a cargo check, it doesn't shows those errors? If cargo check doesn't report anything, so it must be something with your IDE setup.

―
